src/environmentd/src/deployment/preflight.rs

```diff
@@ -190,6 +241,19 @@ pub async fn preflight_0dt(
             // Take over the catalog.
             info!("promoted; attempting takeover");
 
+            // NOTE: There _is_ a window where DDL can happen in the old
```

Comment:
@alex-hunt-materialize If cloud is set up for this, or wants to make that change, we could try and shorten the window for missed DDL a bit more.

Reply:
I don't think we should worry too much about DDL made in the last few seconds before promotion. At most, it will only be a few seconds lost of hydration.